Code/GraphMol/CIPLabeler/CIPMol.cpp

```diff
@@ -0,0 +1,144 @@
+//
+//
+//  Copyright (C) 2020 Schrödinger, LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include <GraphMol/MolOps.h>
+
+#include "CIPMol.h"
+
+namespace RDKit {
+namespace CIPLabeler {
+
+CIPMol::CIPMol(ROMol *mol) : mol{mol} {}
+
+Fraction CIPMol::getFractionalAtomicNum(Atom *atom) const {
+  PRECONDITION(atom, "bad atom")
+  if (atomnums.empty())
+    const_cast<CIPMol *>(this)->atomnums = calcFracAtomNums(this);
+  return atomnums[atom->getIdx()];
+}
+
+unsigned CIPMol::getNumAtoms() const { return mol->getNumAtoms(); }
+
+unsigned CIPMol::getNumBonds() const { return mol->getNumBonds(); };
+
+Atom *CIPMol::getAtom(int idx) const { return mol->getAtomWithIdx(idx); };
+
+CXXAtomIterator<MolGraph, Atom *> CIPMol::atoms() const { return mol->atoms(); }
+
+Bond *CIPMol::getBond(int idx) const { return mol->getBondWithIdx(idx); };
+
+CIPMolIterator<Bond *, ROMol::OEDGE_ITER> CIPMol::getBonds(Atom *atom) const {
+  return {mol, mol->getAtomBonds(atom)};
+}
+
+CIPMolIterator<Atom *, ROMol::ADJ_ITER> CIPMol::getNeighbors(Atom *atom) const {
+  return {mol, mol->getAtomNeighbors(atom)};
+}
+
+bool CIPMol::isInRing(Bond *bond) const {
+  const auto rings = mol->getRingInfo();
+
+  if (!rings->isInitialized()) {
+    MolOps::findSSSR(*mol);
```

Comment:
Since you just care about whether or not an atom is in a ring it would be a ton faster to do: 
```suggestion
    MolOps::fastFindRings(*mol);
```

Reply:
Thanks, @greglandrum, I'll include that in the next update.